lib/rules/indentation/index.js

```diff
@@ -641,31 +641,24 @@ function inferRootIndentLevel (root, baseIndentLevel, indentSize) {
 	}
 
 	let indents = []
-	let foundIndents = root.raws.codeBefore && (/(?:^|\n)([ \t]*)\S/m).exec(root.raws.codeBefore)
+	let foundIndents = root.raws.codeBefore && root.raws.codeBefore.match(/(?:^|\n)([ \t]*)\S/gm)
```

Comment:
I would like to see this as `root.raws.codeBefore?.match(/(?:^|\n)([\t ]*)\S/gm)`

We only support Node 18+ (per `package.json` entry), so using optional chaining operator is safe